sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -903,4 +903,21 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
       }
     }
   }
+
+  test("SPARK-37932: view join view self with having filter") {
+    withTable("t") {
+      withView("v1") {
+        Seq((2, "test2"), (3, "test3"), (1, "test1")).toDF("id", "name")
+          .write.format("parquet").saveAsTable("t")
+        sql("CREATE VIEW v1 (id, name) AS SELECT id, name FROM t")
+
+        sql("""SELECT l1.id FROM v1 l1
+              | INNER JOIN (
+              |   SELECT id FROM v1
+              |   GROUP BY id HAVING COUNT(DISTINCT name) > 1
+              | ) l2 ON l1.id = l2.id GROUP BY l1.name, l1.id;
+        """.stripMargin)
```

Comment:
can we also assert query answer correctness ?

Reply:
OK. I will assert query answer correctness in ut

Reply:
```
"""
  ｜SELECT l1.id FROM v1 l1
  ｜INNER JOIN (
  ｜  SELECT id FROM v1
  ｜  GROUP BY id HAVING COUNT(DISTINCT name) > 1
  ｜) l2 ON l1.id = l2.id
  ｜GROUP BY l1.name, l1.id
  ｜""".stripMargin
```

Reply:
I already assert query answer correctness in UT.